### PR TITLE
Rewrite acpid module in a more generic way

### PR DESCRIPTION
**Pull request by k0ral**
Follow-up of #12441 .

This is an untested draft version. I'm not proficient in nixlang so this is most certainly perfectible.

### CODE REVIEW

**Comment by peti** [comment]
With that patch applied, how do users discovers which events exist? I.e. how do I know there is such as a thing as `cdNextEvent`? The old code defined an explicit option for this which showed up in the manual, but the new code doesn't seem to do that.

File: nixos/modules/services/hardware/acpid.nix [hunk 1]
```diff
@@ -126,58 +36,28 @@ in
...
-        default = "";
-        description = "Shell commands to execute on an cd/next.* event.";
-      };
-
-      cdPrevEventCommands = mkOption {
-        type = types.lines;
-        default = "";
-        description = "Shell commands to execute on an cd/prev.* event.";
+      events = mkOption {
+        type = types.listOf types.optionSet;
```
Comment (globin) [inline nixos/modules/services/hardware/acpid.nix]: This should probably just be an attrSet with `name` as key.

**Comment by edolstra** [comment]
We should probably keep the options that pre-date #12441. However, they can be implemented using `services.acpid.events`.

File: nixos/modules/services/hardware/acpid.nix (hunk 1 above)
Comment (k0ral) [inline nixos/modules/services/hardware/acpid.nix]: Done. Thanks for the hint ! :)

**Comment by k0ral** [comment]
@peti I provided enough examples for users to be able to find out by themselves what are the available events. Though #12441 is better in terms of discoverability, it has the drawback of duplicating `acpid`'s options in NixOS. In other words, it makes the module more dependent on `acpid`'s internals. Isn't that what we want to avoid ?

**Comment by heydojo** [comment]
I'm subscribing to this thread because I am interested in adding laptop lid support using atieventsd and systemd. atieventsd uses /run/acpid.socket and looks like it might be fun :)

**Comment by k0ral** [comment]
@edolstra I've updated the pull request according to your comment, although keeping the old options seems redundant/confusing. I guess your intent is to be backward compatibility ?

**Comment by k0ral** [comment]
Is there anything yet that prevents this pull request from being merged ?

**Pull request merged**